src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -473,6 +486,9 @@ protected void deleteUser(User user) {
         studentScoreRepository.deleteAllByUserId(user.getId());
         learningGoalProgressRepository.deleteAllByUserId(user.getId());
         exerciseHintActivationRepository.deleteAllByUser(user);
+        postRepository.deleteAllByAuthor(user);
+        conversationRepository.deleteAllByCreator(user);
```

Comment:
Should OneToOneChats also be deleted if the user is only a participant, regardless if they are the creator or not?

I assume that other code and especially the client assumes that such a conversation always has _exactly two_ participants and could potentially break here.

Reply:
Good point. I'll make the necessary changes to ensure that they are deleted.